tests/unit/acquisition/function/test_utils.py

```diff
@@ -0,0 +1,143 @@
+        MultivariateNormalCDF(sample_size=num_sobol, dim=dim, dtype=dtype)
+
+
+@pytest.mark.parametrize("num_sobol", [1, 10, 100])
+@pytest.mark.parametrize("dim", [2, 3, 5])
+def test_make_mvn_cdf_raises_exception_for_incorrect_batch_size(
+    num_sobol: int,
+    dim: int,
+) -> None:
+
+    # Set data type and jitter
+    dtype = tf.float64
+    batch_size = 0
+
+    # Draw x randomly
+    x = tf.random.normal((batch_size, dim), dtype=dtype)
+
+    # Draw mean randomly
+    mean = tf.random.normal((batch_size, dim), dtype=dtype)
+
+    # Draw covariance randomly
+    cov = tf.random.normal((batch_size, dim, dim), dtype=dtype)
+    cov = tf.matmul(cov, cov, transpose_a=True)
+
+    with pytest.raises(tf.errors.InvalidArgumentError):
+        MultivariateNormalCDF(sample_size=num_sobol, dim=dim, dtype=dtype)(x=x, mean=mean, cov=cov)
+
+
+@pytest.mark.parametrize("num_sobol", [200])
+@pytest.mark.parametrize("dim", [2, 3, 5])
+@pytest.mark.parametrize("batch_size", [1, 2, 4])
+def test_make_genz_cdf_matches_naive_monte_carlo_on_random_tasks(
+    num_sobol: int,
+    dim: int,
+    batch_size: int,
+) -> None:
+    def mc_mvn_cdf(
+        x: TensorType,
+        mean: TensorType,
+        cov: TensorType,
+        num_samples: int = int(1e6),
+    ) -> TensorType:
+
+        # Define multivariate normal
+        normal = tfd.MultivariateNormalTriL(
+            loc=mean,
+            scale_tril=tf.linalg.cholesky(cov),
+        )
+
+        # Draw samples
+        samples = normal.sample(sample_shape=[num_samples])
+
+        # Check shapes of input tensors
+        tf.debugging.assert_shapes(
+            [
+                (x, ("B", "Q")),
+                (mean, ("B", "Q")),
+                (cov, ("B", "Q", "Q")),
+                (samples, ("S", "B", "Q")),
+            ]
+        )
+
+        # Compute Monte Carlo estimate
+        indicator = tf.reduce_all(tf.math.less(samples, x[None, ...]), axis=-1)
+        mc_mvn_cdf = tf.reduce_mean(tf.cast(indicator, tf.float64), axis=0)
+
+        return mc_mvn_cdf
+
+    # Seed sampling for reproducible testing
+    tf.random.set_seed(0)
+
+    # Set data type and jitter
+    dtype = tf.float64
+    jitter = 1e-6
+
+    # Draw x randomly
+    x = tf.random.normal((batch_size, dim), dtype=dtype) / dim ** 0.5
+
+    # Draw mean randomly
+    mean = tf.random.normal((batch_size, dim), dtype=dtype) / dim ** 0.5
+
+    # Draw covariance randomly
+    cov = tf.random.normal((batch_size, dim, dim), dtype=dtype) / dim ** 0.5
+    cov = tf.matmul(cov, cov, transpose_a=True) + jitter * tf.eye(dim, dtype=dtype)[None, :, :]
+
+    # Set up Genz approximation and direct Monte Carlo estimate
+    genz_cdf = MultivariateNormalCDF(sample_size=num_sobol, dim=dim, dtype=dtype)(
+        x=x, mean=mean, cov=cov
+    )
+    mc_cdf = mc_mvn_cdf(x=x, mean=mean, cov=cov)
+
+    # Check that the Genz and direct Monte Carlo estimates agree
+    tf.debugging.assert_near(mc_cdf, genz_cdf, rtol=5e-1)
```

Comment:
that's a really high tolerance... is it sufficient to test that the CDF code is correct?

Reply:
Yes, this is a very large relative tolerance. I have set it to this value because the naive approximation is very poor for higher dimensions, so the test would otherwise fail (not because the Genz CDF is poor, but because the MCEI approximation is poor).

A solution is to increase the number of samples further, beyond `10^6`. Increasing this to `10^8` allows reducing the tolerance down to `3e-1`, but not much lower. I don't think it's realistic to use a larger amount of samples since this would slow the tests quite a bit. I think it makes sense to stick with `10^8` samples and a tolerance of `3e-1`. What do you think?

Reply:
Thanks for the suggestion! In this case, I think it might be sensible to keep the current test (with randomly sampled problems and relatively loose tolerance), and add one more test like the one you posted above.